eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version3/ChannelCodecs3.scala

```diff
@@ -42,9 +43,26 @@ private[channel] object ChannelCodecs3 {
         ("path" | keyPathCodec) ::
         ("parent" | int64)).as[ExtendedPrivateKey]
 
-    val channelVersionCodec: Codec[ChannelVersion] = bits(ChannelVersion.LENGTH_BITS).as[ChannelVersion]
-
-    def localParamsCodec(channelVersion: ChannelVersion): Codec[LocalParams] = (
+    val channelConfigCodec: Codec[ChannelConfigOptions] = lengthDelimited(bytes).xmap(b => {
+      val activated: Set[ChannelConfigOption] = b.bits.toIndexedSeq.reverse.zipWithIndex.collect {
+        case (true, 0) => ChannelConfigOptions.FundingPubKeyBasedChannelKeyPath
+      }.toSet
+      ChannelConfigOptions(activated)
+    }, cfg => {
+      val indices = cfg.activated.map(_.supportBit)
+      if (indices.isEmpty) {
+        ByteVector.empty
+      } else {
+        // NB: when converting from BitVector to ByteVector, scodec pads right instead of left, so we make sure we pad to bytes *before* setting bits.
+        var buffer = BitVector.fill(indices.max + 1)(high = false).bytes.bits
+        indices.foreach(i => buffer = buffer.set(i))
+        buffer.reverse.bytes
+      }
+    })
```

Comment:
This codec is a bit 🤢, but unfortunately it seems `scodec` doesn't work well with bitfields. 

Since we use bitfields at several places, it's probably worth it to he define a generic byte-aligned, right-to-left, bitfield codec? I'll give it a try.

Reply:
Agreed, it's useful so that we don't unintentionally go back to storing something that isn't byte-aligned!